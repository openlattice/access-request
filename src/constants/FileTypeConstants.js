export const AUDIO_TYPE_PREFIX = 'audio/';
export const IMAGE_TYPE_PREFIX = 'image/';
export const VIDEO_TYPE_PREFIX = 'video/';

export const ALL_AUDIO_MIME_TYPE = `${AUDIO_TYPE_PREFIX}*`;
export const ALL_IMAGE_MIME_TYPE = `${IMAGE_TYPE_PREFIX}*`;
export const ALL_VIDEO_MIME_TYPE = `${VIDEO_TYPE_PREFIX}*`;
export const DOCX_MIME_TYPE = 'application/vnd.openxmlformats-officedocument.wordprocessingml.document';
export const PDF_MIME_TYPE = 'application/pdf';
export const TEXT_CSV = 'text/csv';
export const TEXT_XML = 'text/xml';
export const XLSX_MIME_TYPE = 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet';
export const XML_HEADER = "<?xml version='1.0' encoding='UTF-8'?>";
