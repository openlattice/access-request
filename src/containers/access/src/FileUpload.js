// @flow
import React from 'react';

import Dropzone from '../../../components/Dropzone';

type Props = {
  onChange :Function
}

const FileUpload = ({
  onChange
} :Props) => {

  const onDrop = (files) => {
    files.forEach((file) => {
      const { name, type } = file;
      const reader = new FileReader();

      reader.onload = (event) => {
        const base64 = event.target.result;

        onChange({
          file: {
            base64,
            name,
            type
          }
        });

      };
      reader.readAsDataURL(file);
    });
  };

  return (
    <Dropzone onDrop={onDrop} />
  );
};

export default FileUpload;
