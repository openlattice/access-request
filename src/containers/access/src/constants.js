// @flow
const CONSENT_FORM :'Consent Form' = 'Consent Form';
const CPHS_REVIEW :'CPHS Review' = 'CPHS Review';
const DATA_RELEASE_LETTER :'Data Release Letter' = 'Data Release Letter';
const DATA_SECURITY_LETTER :'Data Security Letter' = 'Data Security Letter';
const DATA_USE_AGREEMENT :'Data Use Agreement' = 'Data Use Agreement';
const DEPARTMENT_SUPPORT_LETTER :'Department Support Letter' = 'Department Support Letter';
const IRB_APPROVAL :'IRB Approval' = 'IRB Approval';
const PARTICIPANTS_BOR :"Participant's Bill of Rights" = "Participant's Bill of Rights";

const TAG_OPTIONS = [
  { label: CONSENT_FORM, value: CONSENT_FORM },
  { label: CPHS_REVIEW, value: CPHS_REVIEW },
  { label: DATA_RELEASE_LETTER, value: DATA_RELEASE_LETTER },
  { label: DATA_SECURITY_LETTER, value: DATA_SECURITY_LETTER },
  { label: DATA_USE_AGREEMENT, value: DATA_USE_AGREEMENT },
  { label: DEPARTMENT_SUPPORT_LETTER, value: DEPARTMENT_SUPPORT_LETTER },
  { label: IRB_APPROVAL, value: IRB_APPROVAL },
  { label: PARTICIPANTS_BOR, value: PARTICIPANTS_BOR },
];

export {
  CONSENT_FORM,
  CPHS_REVIEW,
  DATA_RELEASE_LETTER,
  DATA_SECURITY_LETTER,
  DATA_USE_AGREEMENT,
  DEPARTMENT_SUPPORT_LETTER,
  IRB_APPROVAL,
  PARTICIPANTS_BOR,
  TAG_OPTIONS,
};
