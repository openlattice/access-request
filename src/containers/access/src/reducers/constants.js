export const ACCESS_REQUEST = 'accessRequest';
export const ATTACHMENTS = 'attachments';
export const FORMS = 'forms';
