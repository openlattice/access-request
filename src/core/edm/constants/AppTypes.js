/*
 * @flow
 */

import { Models } from 'lattice';

const { FQN } = Models;

const APP_TYPE_FQNS = {
  FORM: FQN.of('app.form'),
  ACCESS_REQUEST_SUBMISSION: FQN.of('app.accessrequestsubmission'),
};

export default APP_TYPE_FQNS;
