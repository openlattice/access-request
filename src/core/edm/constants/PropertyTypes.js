/*
 * @flow
 */

import { Models } from 'lattice';

const { FQN } = Models;

const PROPERTY_TYPE_FQNS = {
  ID: FQN.of('ol.id'),
  RJSF_JSON_SCHEMA: FQN.of('ol.rjsfjsonschema'),
  RJSF_UI_SCHEMA: FQN.of('ol.rjsfuischema'),
  FORM_DATA: FQN.of('ol.formdata'),
  REQUEST_DATE_TIME: FQN.of('ol.requestdatetime'),
  TYPE: FQN.of('ol.type'),
};

export default PROPERTY_TYPE_FQNS;
