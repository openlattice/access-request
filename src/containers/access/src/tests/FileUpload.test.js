import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';

import { MOCK_PDF_FILE } from './constants';

import FileUpload from '../FileUpload';

describe('FileUpload', () => {
  const mockOnChange = jest.fn();
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('match snapshot', () => {
    const wrapper = mount(<FileUpload onChange={mockOnChange} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('onDrop invoke readAsDataUrl', async () => {
    const wrapper = mount(<FileUpload onChange={mockOnChange} />);
    const readAsDataURLSpy = jest.spyOn(FileReader.prototype, 'readAsDataURL');

    const inputWrapper = wrapper.find('input');
    await act(async () => {
      inputWrapper.simulate('change', {
        target: { files: [MOCK_PDF_FILE] }
      });
      await new Promise((resolve) => setTimeout(resolve, 100));
      wrapper.update();
    });
    expect(readAsDataURLSpy).toHaveBeenCalledTimes(1);
    expect(mockOnChange).toHaveBeenCalledTimes(1);
    expect(mockOnChange.mock.calls[0][0]).toHaveProperty('file');
    expect(mockOnChange.mock.calls[0][0]).toHaveProperty(['file', 'base64']);
    expect(mockOnChange.mock.calls[0][0]).toHaveProperty(['file', 'name']);
    expect(mockOnChange.mock.calls[0][0]).toHaveProperty(['file', 'type']);
  });

});
