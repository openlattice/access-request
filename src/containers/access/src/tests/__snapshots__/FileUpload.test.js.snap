// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUpload match snapshot 1`] = `
<FileUpload
  onChange={[MockFunction]}
>
  <Dropzone
    onDrop={[Function]}
  >
    <Dropzone__DropzoneWrapper
      isDragAccept={false}
      isDragActive={false}
      isDragReject={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragEnter={[Function]}
      onDragLeave={[Function]}
      onDragOver={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
    >
      <div
        className="Dropzone__DropzoneWrapper-j9e9j-0 fpLcaF"
        onBlur={[Function]}
        onClick={[Function]}
        onDragEnter={[Function]}
        onDragLeave={[Function]}
        onDragOver={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
      >
        <input
          accept="image/*,application/pdf,application/vnd.openxmlformats-officedocument.wordprocessingml.document"
          autoComplete="off"
          multiple={true}
          onChange={[Function]}
          onClick={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
          tabIndex={-1}
          type="file"
        />
        <ForwardRef(WithStyles)
          component="span"
        >
          <ForwardRef(WithStyles)
            classes={
              Object {
                "body1": "jss9",
                "body2": "jss10",
                "button": "jss12",
                "caption": "jss11",
                "h1": "jss1",
                "h2": "jss2",
                "h3": "jss3",
                "h4": "jss4",
                "h5": "jss5",
                "h6": "jss6",
                "overline": "jss13",
                "subtitle1": "jss7",
                "subtitle2": "jss8",
              }
            }
            component="span"
          >
            <ForwardRef(Typography)
              classes={
                Object {
                  "alignCenter": "MuiTypography-alignCenter",
                  "alignJustify": "MuiTypography-alignJustify",
                  "alignLeft": "MuiTypography-alignLeft",
                  "alignRight": "MuiTypography-alignRight",
                  "body1": "MuiTypography-body1 jss9",
                  "body2": "MuiTypography-body2 jss10",
                  "button": "MuiTypography-button jss12",
                  "caption": "MuiTypography-caption jss11",
                  "colorError": "MuiTypography-colorError",
                  "colorInherit": "MuiTypography-colorInherit",
                  "colorPrimary": "MuiTypography-colorPrimary",
                  "colorSecondary": "MuiTypography-colorSecondary",
                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                  "displayBlock": "MuiTypography-displayBlock",
                  "displayInline": "MuiTypography-displayInline",
                  "gutterBottom": "MuiTypography-gutterBottom",
                  "h1": "MuiTypography-h1 jss1",
                  "h2": "MuiTypography-h2 jss2",
                  "h3": "MuiTypography-h3 jss3",
                  "h4": "MuiTypography-h4 jss4",
                  "h5": "MuiTypography-h5 jss5",
                  "h6": "MuiTypography-h6 jss6",
                  "noWrap": "MuiTypography-noWrap",
                  "overline": "MuiTypography-overline jss13",
                  "paragraph": "MuiTypography-paragraph",
                  "root": "MuiTypography-root",
                  "srOnly": "MuiTypography-srOnly",
                  "subtitle1": "MuiTypography-subtitle1 jss7",
                  "subtitle2": "MuiTypography-subtitle2 jss8",
                }
              }
              component="span"
            >
              <span
                className="MuiTypography-root MuiTypography-body1 jss9"
              >
                Drop files here or click to select.
              </span>
            </ForwardRef(Typography)>
          </ForwardRef(WithStyles)>
        </ForwardRef(WithStyles)>
      </div>
    </Dropzone__DropzoneWrapper>
  </Dropzone>
</FileUpload>
`;
